internal_ws/ykcompile/src/arch/x86_64/mod.rs

```diff
@@ -237,7 +237,12 @@ macro_rules! binop_mul_div {
                 },
                 Location::Mem(..) => todo!(),
                 Location::Const { val, .. } => {
-                    if size > 1 {
+                    // Set unused extended operand space to zero.
+                    if size == 1 {
+                        dynasm!(self.asm
+                            ; and ax, 0xff
```

Comment:
How is this setting `ax` to zero? Doesn't this just leave `ax` as is?

Reply:
I was about to say:

> It sets the top byte of AX to zero. In the `r/m8` variants, the first operand is all of AX, but we only care for the lower byte.

But, that's only true of `DIV` I now realise...

![image](https://user-images.githubusercontent.com/604955/109956459-94bc9100-7cdb-11eb-87f4-823d9871bbcc.png)

![image](https://user-images.githubusercontent.com/604955/109956697-dea57700-7cdb-11eb-8871-0d3f628e96b7.png)

Let me see if I can find a way to only do this for DIV without a string comparison at runtime.

Reply:
Of course, the obvious fix causes test failures...

Reply:
Fixed in 549d394cd.